holidays/utils.py

```diff
@@ -435,3 +437,44 @@ def list_supported_financial(include_aliases: bool = True) -> dict[str, list[str
         subdivision codes.
     """
     return _list_supported_entities(EntityLoader.get_financial_codes(include_aliases))
+
+
+def list_long_weekends(
+    instance: HolidayBase,
+    include_non_long_weekends: Optional[bool] = False,
+) -> list:
```

Comment:
Consider unhardcoded the minimum lengths + adding docstrings

I'm not that sure about the current `include_non_long_weekends` name as it's not that clear at the first glance - any ideas? @KJhellico @arkid15r 

Maybe `require_weekend_overlap` or something similar
```suggestion
def list_long_weekends(
    instance: HolidayBase, minimum_holiday_length: int = 3, include_non_long_weekends: bool = False
) -> list:
    """Get all long consecutive holidays.
    
    Args:
        instance:
            `HolidaysBase` object containing holiday data.

        minimum_length:
            The minimum number of consecutive days required for a holiday period
            to be considered a long weekend. Defaults to 3.

        include_non_long_weekends:
            Whether to include consecutive holidays that do not contain any weekend days.
            Defaults to False.
            
    Returns:
        A list of long consecutive holiday periods longer than or equal
        to the specified minimum length.
    """
```

Reply:
Yes, please do